core/array.cpp

```diff
@@ -204,9 +204,12 @@ bool Array::has(const Variant& p_value) const {
 	return _p->array.find(p_value, 0) != -1;
 }
 
-void Array::remove(int p_pos) {
+Variant Array::remove(int p_pos) {
 
+	Variant ret = _p->array.get(p_pos);
```

Comment:
I'm still learning how things work, but now that I look at it I think this is the problem. Because the error is in `vector.h` **not** `dvector.h`. But I still don't understand why this would happen. I thoudl have thought that using `array.get(p_pos)` with `p_pos` outside of range will automatically give an error in the debugger. Also I would of thought (now) that `ERR_FAIL_INDEX_V(p_index, s, T());` back in `dvector.h` would be useless because of the same situation, using `get()` before. Anyway, I'm gonna let the masters take over! Thanks